components/constellation/constellation.rs

```diff
@@ -4155,9 +4156,14 @@ where
         debug!("Setting activity of {} to be {:?}.", pipeline_id, activity);
         if let Some(pipeline) = self.pipelines.get(&pipeline_id) {
             pipeline.set_activity(activity);
+            let media = ServoMedia::get().unwrap();
+            let client_context_id =
+                ClientContextId::build(pipeline_id.namespace_id.0, pipeline_id.index.0.get());
             let child_activity = if activity == DocumentActivity::Inactive {
+                media.suspend(&client_context_id);
```

Comment:
Have you considered doing the `suspend` and `resume` operations in script when the document activity changes? I think that might be more robust to required changes to media to fully support multiprocess. 

You could move those operations to https://github.com/servo/servo/blob/7b653cad7b0d891e06e4847ad738f38a9510dd92/components/script/dom/document.rs#L496

and also https://github.com/servo/servo/blob/7b653cad7b0d891e06e4847ad738f38a9510dd92/components/script/script_thread.rs#L3254 (since a document in the session history might require reloading)

Reply:
Good point! I'm applying your suggestion. Thanks